src/setup-java.ts

```diff
@@ -16,6 +17,14 @@ async function run() {
 
     const matchersPath = path.join(__dirname, '..', '.github');
     console.log(`##[add-matcher]${path.join(matchersPath, 'java.json')}`);
+
+    const id = core.getInput('server-id', {required: false});
+    const username = core.getInput('username', {required: false});
+    const password = core.getInput('password', {required: false});
```

Comment:
You might want to set `password` as a secret to protect users from accidental password leakage (in case they're populating this input without a SECRET already). This will enable masking for logs.

```ts
core.setSecret(password);
```
See https://github.com/actions/toolkit/tree/master/packages/core#setting-a-secret

Reply:
If they are storing the password in the file unencrypted it makes no difference if it is masked in the logs.

Reply:
And... someone only now shared with me that we can use env variables within the `settings.xml` file. https://stackoverflow.com/a/31251518

So a generic settings file like so might work and we won't need to save the token into it.

```xml
<servers>
    <server>
      <id>github</id>
      <username>${env.GITHUB_ACTOR}</username>
      <password>${env.GITHUB_TOKEN}</password>
    </server>
</servers>
```